### DISCUSSION TITLE
Segmentation fault in coefficient form pde toolbox in Mesh generation

### DISCUSSION THREAD

**Discussion by lotus-prog** [feelpp/feelpp#1592] [Toolboxes] [answered]
Hello @prudhomm , 
I had a segmentation fault error when I execute the command feelpp_toolbox_coefficientformpdes --config-file square2d.cfg despite that I'm in  the directory csmi-edp/feelpp/toolboxes/coefficientformpdes/square. 
Could you help me please ? 

[![feelpp.png](https://i.postimg.cc/8kYtwyV8/feelpp.png)](https://postimg.cc/0rpYy0QZ)
[![feelpp2.png](https://i.postimg.cc/SKmrj3c9/feelpp2.png)](https://postimg.cc/nXw7PdRV)

**Comment by prudhomm** [comment]
Could you upload the image in the discussion?
The links don't work for me
Simply edit your question and upload the images @lotus-prog

**Reply by prudhomm** [reply]
Also the language is English

**Reply by lotus-prog** [reply]
Done

**Reply by prudhomm** [reply]
@lotus-prog could you provide the .geo ? you are using gmsh 4 right ? (`gmsh --version`) 
and the .cfg

**Reply by lotus-prog** [reply]
Here it is

**Comment by lotus-prog** [comment]
[![feelpp3.png](https://i.postimg.cc/85MCFpMk/feelpp3.png)](https://postimg.cc/R6Cmyzg2)

**Reply by prudhomm** [reply]
could you remove `$HOME/feelppdb/toolboxes/coefficientformpdes` and re-run the code ?

**Reply by lotus-prog** [reply]
with Delete Permanently coeficientformpdes ?

**Reply by StevedeRose** [reply]
Hello @prudhomm,
I have the exact same issue. (I tried on two different computers)

**Reply by prudhomm** [reply]
Ok do you have the variable `LC_CTYPE` oe `LC_<xxx>` or `LANG`  in your environment?

Type `env` to list them

**Reply by prudhomm** [reply]
If yes undefine them

Eg

```
unset LC_CTYPE
```
Then re run the feelpp app

**Reply by 5TuX** [reply]
I also had the same issue, I found a LANG variable and unset it and now it works. Thanks!

**Reply by StevedeRose** [reply]
It worked. Thanks.

**Comment by melissauntz** [comment]
Hello @prudhomm, I had an issue which looked like the error described above but not exactly the same (SIGABRT received). I tried to unset the LANG variable but now I have a Segmentation fault :
![vscode](https://user-images.githubusercontent.com/77633048/112370354-90353800-8cdd-11eb-8f78-f50b0115213a.png)

**Reply by 5TuX** [reply]
Did you find variables starting with `LC_` in your environment ? You can copy/paste the result of `env` in a file to search more easily

**Reply by melissauntz** [reply]
I searched, but only had a LANG variable..

**Reply by melissauntz** [reply]
These lines appeared after undefined LANG : 
![erase](https://user-images.githubusercontent.com/77633048/112371792-4fd6b980-8cdf-11eb-985c-32b34bd09498.png)

**Reply by prudhomm** [reply]
could you provide a complete screenshot @melissauntz  ?

**Reply by melissauntz** [reply]
I already published it above : (I unset LANG and did not find LC_.... variable)
![github](https://user-images.githubusercontent.com/77633048/112381920-76025680-8ceb-11eb-8a58-0c073201ec19.png)

**Reply by melissauntz** [reply]
@prudhomm  Would it be useful to send you the results of `env` ?

**Comment by prudhomm** [comment, marked as answer]
The issue is that Gmsh does not support well internationalisation and in particular the environment variables associated to them like `LANG`, `LC_CTYPE`, ....  You should unset them before running the feelpp applications if the mesh needs to be generated.

for example
```
unset LC_CTYPE LANG 
```

**Reply by lotus-prog** [reply]
It doesn't work for me, I did the instruction wrote earlier ( because I had LANG ) and now I have the same issue than Melissa
